### ISSUE TITLE
unable to install scGate or ProjecTILs libraries

### ISSUE THREAD

**Issue by ashkanshahbandi** [carmonalab/ProjecTILs#41]
Hi, I have been trying to install these libraries into R-Studio, but I keep getting the following errors:
> remotes::install_github("carmonalab/scGate")
Downloading GitHub repo carmonalab/scGate@HEAD
These packages have more recent versions available.
It is recommended to update all of them.
Which would you like to update?

1: All                            
2: CRAN packages only             
3: None                           
4: stringi (1.7.6 -> 1.7.8) [CRAN]

Enter one or more numbers, or an empty line to skip updates: 1
stringi (1.7.6 -> 1.7.8) [CRAN]
Skipping 1 packages not available: UCell
Installing 2 packages: stringi, UCell
Installing packages into ‘C:/Users/ashka/Documents/renv/library/R-4.1/x86_64-w64-mingw32’
(as ‘lib’ is unspecified)

  There is a binary version available but the source version is later:
        binary source needs_compilation
stringi  1.7.6  1.7.8              TRUE

  Binaries will be installed
trying URL 'https://cloud.r-project.org/bin/windows/contrib/4.1/stringi_1.7.6.zip'
Content type 'application/zip' length 16449819 bytes (15.7 MB)
downloaded 15.7 MB

package ‘stringi’ successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\ashka\AppData\Local\Temp\Rtmpc50GBN\downloaded_packages
Running `R CMD build`...
* checking for file 'C:\Users\ashka\AppData\Local\Temp\Rtmpc50GBN\remotes73b047e96ee2\carmonalab-scGate-1b23084/DESCRIPTION' ... OK
* preparing 'scGate':
* checking DESCRIPTION meta-information ... OK
* checking for LF line-endings in source and make files and shell scripts
* checking for empty or unneeded directories
* building 'scGate_1.2.0.tar.gz'
Installing package into ‘C:/Users/ashka/Documents/renv/library/R-4.1/x86_64-w64-mingw32’
(as ‘lib’ is unspecified)
ERROR: dependency 'UCell' is not available for package 'scGate'
* removing 'C:/Users/ashka/Documents/renv/library/R-4.1/x86_64-w64-mingw32/scGate'
Warning messages:
1: In untar2(tarfile, files, list, exdir, restore_times) :
  skipping pax global extended headers
2: In untar2(tarfile, files, list, exdir, restore_times) :
  skipping pax global extended headers
3: package ‘UCell’ is not available for this version of R

A version of this package for your version of R might be available elsewhere,
see the ideas at
https://cran.r-project.org/doc/manuals/r-patched/R-admin.html#Installing-packages 
4: In i.p(...) :
  installation of package ‘C:/Users/ashka/AppData/Local/Temp/Rtmpc50GBN/file73b0491424d0/scGate_1.2.0.tar.gz’ had non-zero exit status

> remotes::install_github("carmonalab/ProjecTILs")
Downloading GitHub repo carmonalab/ProjecTILs@HEAD
These packages have more recent versions available.
It is recommended to update all of them.
Which would you like to update?

1: All                            
2: CRAN packages only             
3: None                           
4: stringi (1.7.6 -> 1.7.8) [CRAN]

Enter one or more numbers, or an empty line to skip updates: 
Running `R CMD build`...
* checking for file 'C:\Users\ashka\AppData\Local\Temp\Rtmpc50GBN\remotes73b014782336\carmonalab-ProjecTILs-370abd8/DESCRIPTION' ... OK
* preparing 'ProjecTILs':
* checking DESCRIPTION meta-information ... OK
* checking for LF line-endings in source and make files and shell scripts
* checking for empty or unneeded directories
* building 'ProjecTILs_3.0.0.tar.gz'
Installing package into ‘C:/Users/ashka/Documents/renv/library/R-4.1/x86_64-w64-mingw32’
(as ‘lib’ is unspecified)
ERROR: dependencies 'UCell', 'scGate' are not available for package 'ProjecTILs'
* removing 'C:/Users/ashka/Documents/renv/library/R-4.1/x86_64-w64-mingw32/ProjecTILs'
Warning messages:
1: In untar2(tarfile, files, list, exdir, restore_times) :
  skipping pax global extended headers
2: In untar2(tarfile, files, list, exdir, restore_times) :
  skipping pax global extended headers
3: In i.p(...) :
  installation of package ‘C:/Users/ashka/AppData/Local/Temp/Rtmpc50GBN/file73b07ff54eff/ProjecTILs_3.0.0.tar.gz’ had non-zero exit status

I have been able to successfully install the packages on another computer a few months ago, so I'm not sure what the problem is now. Can someone help me with this? Thanks!

**Comment by mass-a** [comment]
Hello, I can see from your log:
```
Skipping 1 packages not available: UCell
```
What version of R and Bioconductor are you using? Note that `UCell` is available from Bioconductor starting from Bioc-3.15 (R-4.2).

If you are using R<=4.1 you should install an older version of UCell from the github repo:
```
library(remotes)
remotes::install_github("carmonalab/UCell", ref="v1.3")
```

Similarly, `scGate` should be installed from github:
```
remotes::install_github("carmonalab/scGate")
```

Best
-m

**Comment by ashkanshahbandi** [comment]
> Hello, I can see from your log:
> 
> ```
> Skipping 1 packages not available: UCell
> ```
> 
> What version of R and Bioconductor are you using? Note that `UCell` is available from Bioconductor starting from Bioc-3.15 (R-4.2).
> 
> If you are using R<=4.1 you should install an older version of UCell from the github repo:
> 
> ```
> library(remotes)
> remotes::install_github("carmonalab/UCell", ref="v1.3")
> ```
> 
> Similarly, `scGate` should be installed from github:
> 
> ```
> remotes::install_github("carmonalab/scGate")
> ```
> 
> Best -m

Hi, I am using R.4.2.1


I installed scGate from github but I am now getting the following errors:

> remotes::install_github("carmonalab/scGate")
Downloading GitHub repo carmonalab/scGate@HEAD
Installing 1 packages: GenomeInfoDbData
Installing package into ‘C:/Users/ashka/Documents/renv/library/R-4.2/x86_64-w64-mingw32’
(as ‘lib’ is unspecified)
installing the source package ‘GenomeInfoDbData’

trying URL 'https://bioconductor.org/packages/3.15/data/annotation/src/contrib/GenomeInfoDbData_1.2.8.tar.gz'
Content type 'application/x-gzip' length 11512907 bytes (11.0 MB)
downloaded 11.0 MB

* installing *source* package 'GenomeInfoDbData' ...
** using staged installation
** data
** inst
** help
*** installing help indices
** building package indices
Error in file(filename, "r", encoding = encoding) : 
  cannot open the connection
Calls: source -> file
In addition: Warning message:
In file(filename, "r", encoding = encoding) :
  cannot open file 'renv/activate.R': No such file or directory
Execution halted
ERROR: installing package indices failed
* removing 'C:/Users/ashka/Documents/renv/library/R-4.2/x86_64-w64-mingw32/GenomeInfoDbData'

The downloaded source packages are in
	‘C:\Users\ashka\AppData\Local\Temp\RtmpwpIrnt\downloaded_packages’
Running `R CMD build`...
* checking for file 'C:\Users\ashka\AppData\Local\Temp\RtmpwpIrnt\remotes436879ff78e2\carmonalab-scGate-1b23084/DESCRIPTION' ... OK
* preparing 'scGate':
* checking DESCRIPTION meta-information ... OK
* checking for LF line-endings in source and make files and shell scripts
* checking for empty or unneeded directories
* building 'scGate_1.2.0.tar.gz'
Installing package into ‘C:/Users/ashka/Documents/renv/library/R-4.2/x86_64-w64-mingw32’
(as ‘lib’ is unspecified)
* installing *source* package 'scGate' ...
** using staged installation
** R
** data
*** moving datasets to lazyload DB
** byte-compile and prepare package for lazy loading
Error in file(filename, "r", encoding = encoding) : 
  cannot open the connection
Calls: source -> file
In addition: Warning message:
In file(filename, "r", encoding = encoding) :
  cannot open file 'renv/activate.R': No such file or directory
Execution halted
ERROR: lazy loading failed for package 'scGate'
* removing 'C:/Users/ashka/Documents/renv/library/R-4.2/x86_64-w64-mingw32/scGate'
Warning messages:
1: In untar2(tarfile, files, list, exdir, restore_times) :
  skipping pax global extended headers
2: In untar2(tarfile, files, list, exdir, restore_times) :
  skipping pax global extended headers
3: In i.p(...) :
  installation of package ‘GenomeInfoDbData’ had non-zero exit status
4: In i.p(...) :
  installation of package ‘C:/Users/ashka/AppData/Local/Temp/RtmpwpIrnt/file43687e8c501d/scGate_1.2.0.tar.gz’ had non-zero exit status

**Comment by mass-a** [comment]
Now your log says that the installation of the `GenomeInfoDbData` package is failing. Perhaps try to install that package manually?
In any case I am not sure why it's being installed, since it is not a dependency of scGate. You also have warnings from `renv`, so perhaps you have to check if you set up your R project correctly.
